src/phoenix/server/api/subscriptions.py

```diff
@@ -63,42 +97,309 @@ async def chat_completion(
+                    revision=revision,
+                    spans=spans,
+                )
+                for revision in revisions
+            ]
+        ):
+            yield payload
+
+        async with info.context.db() as session:
+            if (
+                playground_project_id := await session.scalar(
+                    select(models.Project.id).where(models.Project.name == PLAYGROUND_PROJECT_NAME)
+                )
+            ) is None:
+                playground_project_id = await session.scalar(
+                    insert(models.Project)
+                    .returning(models.Project.id)
+                    .values(
+                        name=PLAYGROUND_PROJECT_NAME,
+                        description="Traces from prompt playground",
+                    )
+                )
+            db_spans = {
+                example_id: span.add_to_session(session, playground_project_id)
+                for example_id, span in spans.items()
+            }
+            assert (
+                dataset_name := await session.scalar(
+                    select(models.Dataset.name).where(models.Dataset.id == dataset_id)
+                )
+            ) is not None
+            if version_id is None:
+                resolved_version_id = await session.scalar(
+                    select(models.DatasetVersion.id)
+                    .where(models.DatasetVersion.dataset_id == dataset_id)
+                    .order_by(models.DatasetVersion.id.desc())
+                    .limit(1)
+                )
+            else:
+                resolved_version_id = await session.scalar(
+                    select(models.DatasetVersion.id).where(
+                        and_(
+                            models.DatasetVersion.dataset_id == dataset_id,
+                            models.DatasetVersion.id == version_id,
+                        )
+                    )
+                )
+            assert resolved_version_id is not None
+            resolved_version_node_id = GlobalID(DatasetVersion.__name__, str(resolved_version_id))
+            experiment = models.Experiment(
+                dataset_id=from_global_id_with_expected_type(input.dataset_id, Dataset.__name__),
+                dataset_version_id=resolved_version_id,
+                name=input.experiment_name or _DEFAULT_PLAYGROUND_EXPERIMENT_NAME,
+                description=input.experiment_description
+                or _default_playground_experiment_description(dataset_name=dataset_name),
+                repetitions=1,
+                metadata_=input.experiment_metadata
+                or _default_playground_experiment_metadata(
+                    dataset_name=dataset_name,
+                    dataset_id=input.dataset_id,
+                    version_id=resolved_version_node_id,
+                ),
+                project_name=PLAYGROUND_PROJECT_NAME,
+            )
+            session.add(experiment)
+            await session.flush()
```

Comment:
do we need this flush here?

Reply:
We need to access `experiment.id` when creating the experiment run objects below, which is populated by the DB